projects/components/src/filtering/filter/builder/types/abstract-filter-builder.ts

```diff
@@ -0,0 +1,38 @@
+import { collapseWhitespace } from '@hypertrace/common';
+import { Filter, FilterOperator, toUrlFilterOperator } from '../../filter-api';
+import { FilterAttribute } from '../../filter-attribute';
+import { FilterAttributeType } from '../../filter-attribute-type';
+
+export abstract class AbstractFilterBuilder<TValue> {
+  public static supportedAttributeTypes: FilterAttributeType[];
+  public static supportedOperators: FilterOperator[];
+
+  protected abstract supportedAttributeTypes(): FilterAttributeType[];
+  protected abstract supportedOperators(): FilterOperator[];
+
+  protected abstract convertStringToValue(value: string): TValue | undefined;
+  protected abstract convertValueToString(value: TValue): string;
+
+  public buildFilters(attribute: FilterAttribute, value: TValue): Filter<TValue>[] {
```

Comment:
Naming on this one - presuming this is used to generate potential choices for auto complete/cell filtering, but without looking at the implementation that's not obvious. Maybe `buildPotentialFilters` or `buildFilterForEachOperator`

Reply:
Yep! I'll change locally.